lib/stateView/document/fetchDocumentsFactory.js

```diff
@@ -15,7 +21,31 @@ function fetchDocumentsFactory(createSVDocumentRepository) {
    */
   async function fetchDocuments(contractId, type, options, stateViewTransaction = undefined) {
     const svDocumentRepository = createSVDocumentRepository(contractId, type);
+
+    let validationResult;
+    const svContract = await svContractMongoDbRepository.find(contractId);
+    if (!svContract) {
+      return [];
+    }
+
+    try {
+      const dataContract = svContract.getDataContract();
+      const documentSchema = dataContract.getDocumentSchema(type);
+      const dataContractIndexFields = getIndexedFieldsFromDocumentSchema(documentSchema);
+
+      validationResult = validateIndexedFields(dataContractIndexFields, options);
+    } catch (e) {
+      if (e instanceof InvalidDocumentTypeError) {
+        return [];
+      }
```

Comment:
What if other error happened? Could we just check if this type is defined without `try/catch` block?